src/main/python/inverse_parser/_node_restorer.py

```diff
@@ -0,0 +1,684 @@
+        xml_node_value = _XmlNodeChildrenGetter.get_unique_child(xml_node)
+        if xml_node_value is not None:
+            py_node.value = _NodeRestorer.restore(xml_node_value)
+        else:
+            py_node.value = None
+
+    @staticmethod
+    def restore_ast_global(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node, with_tag='identifier')
+        py_node.names = [elem.attrib['value']
+                         for elem in xml_node_children]
+
+    @staticmethod
+    def restore_ast_nonlocal(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node, with_tag='identifier')
+        py_node.names = [elem.attrib['value']
+                         for elem in xml_node_children]
+
+    @staticmethod
+    def restore_ast_class_def(xml_node: ET.Element, py_node: ast.AST) -> None:
+        py_node.name = xml_node.attrib['value']
+
+        _AttributeSetter.set_list_field(xml_node, py_node, 'bases')
+        _AttributeSetter.set_list_field(xml_node, py_node, 'keywords')
+        _AttributeSetter.set_list_field(xml_node, py_node, 'body')
+        _AttributeSetter.set_list_field(xml_node, py_node, 'decorator_list')
+
+    # - Async and await:
+
+    @staticmethod
+    def restore_ast_async_function_def(xml_node: ET.Element, py_node: ast.AST) -> None:
+        _NodeRestorer.restore_ast_function_def(xml_node, py_node)
+
+    @staticmethod
+    def restore_ast_await(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_child = _XmlNodeChildrenGetter.get_unique_child(xml_node)
+        py_node.value = _NodeRestorer.restore(xml_node_child)
+
+    @staticmethod
+    def restore_ast_async_for(xml_node: ET.Element, py_node: ast.AST) -> None:
+        _NodeRestorer.restore_ast_for(xml_node, py_node)
+
+    @staticmethod
+    def restore_ast_async_with(xml_node: ET.Element, py_node: ast.AST) -> None:
+        _NodeRestorer.restore_ast_with(xml_node, py_node)
+
+    # - Top level nodes:
+
+    @staticmethod
+    def restore_ast_module(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.body = _NodeRestorer.restore_many(xml_node_children)
+
+    @staticmethod
+    def restore_ast_interactive(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_children = _XmlNodeChildrenGetter.get_children(xml_node)
+        py_node.body = _NodeRestorer.restore_many(xml_node_children)
+
+    @staticmethod
+    def restore_ast_expression(xml_node: ET.Element, py_node: ast.AST) -> None:
+        xml_node_child = _XmlNodeChildrenGetter.get_unique_child(xml_node)
+        py_node.body = _NodeRestorer.restore(xml_node_child)
+
+    # - Corner cases:
+
+    @staticmethod
+    def restore_nonetype(xml_node: ET.Element, py_node: ast.AST) -> None:
+        pass
```

Comment:
Is it ok? Should we implement it? If it is not, describe why in a comment, please

Reply:
Yes, it's ok. This is the valid implementation